Dockerfile.windows

```diff
@@ -250,18 +256,89 @@ RUN `
   Remove-Item C:\gitsetup.zip; `
   `
   Write-Host INFO: Creating source directory...; `
-  New-Item -ItemType Directory -Path ${GOPATH}\src\github.com\docker\docker | Out-Null; `
+  #TODO - Per https://docs.docker.com/engine/reference/builder/ 
+  #       the below line is redundant and should be removed.
+  #New-Item -ItemType Directory -Path ${GOPATH}\src\github.com\docker\docker | Out-Null; `
   `
   Write-Host INFO: Configuring git core.autocrlf...; `
-  C:\git\cmd\git config --global core.autocrlf true; `
+  C:\git\cmd\git config --global core.autocrlf true;
+
+# Ensure all directories exist that we will require below....
+WORKDIR ${GOPATH}\${DOCKER_SRC_REL_LOCATION}\${BUNDLES_REL_LOCATION}
+
+WORKDIR ${GOPATH}
+
+RUN `
+  Function Save-FileIfNonZero() { `
+    Param( `
+      [Parameter(Mandatory=$True, `
+       ParameterSetName="""LogName""")] `
+       $LogName `
+    ) `
+    if (Test-Path $LogName) `
+    { `
+      if ((Get-Item $LogName).length -gt 0) { `
+        $SrcPath=Join-Path -Path $Env:GOPATH -ChildPath $Env:DOCKER_SRC_REL_LOCATION; `
+        $BundlesPath=Join-Path -Path $SrcPath -ChildPath $Env:BUNDLES_REL_LOCATION; `
+        $BundlesMiscPath=Join-Path -Path $BundlesPath -ChildPath 'misc'; `
+        Move-Item -Path $LogName -Destination $BundlesMiscPath; `
+      } else { `
+        Remove-Item $LogName; `
+      } `
+    } `
+  } `
+  Function Build-GoTestSum() { `
```

Comment:
The scripts that we're writing in the Dockerfile now are a bit lengthy, and more complex to maintain (having to keep all the line-continuations (```) into account.

Perhaps we should put these function in a `.ps` file (easier to write/maintain), and;

Somewhere at the top of the Dockerfile (assuming the functions themself don't change often);

```Dockerfile
COPY hack/dockerfile/install/gotestsum.installer.ps1 hack/dockerfile/install/gotestsum.installer.ps1
```

And later on, call those functions


```Dockerfile
RUN . "hack/dockerfile/install/gotestsum.installer.ps1" Build-GoTestSum 
```

(or however that should be done in PowerShell)

Reply:
Good suggestion and something I have put on my plate. It will be done in a separate PR